src/services/thumbnail/service.ts

```diff
@@ -1,27 +1,33 @@
 import path from 'path';
 import sharp from 'sharp';
 import { Readable } from 'stream';
+import { injectable } from 'tsyringe';
 
 import FileService from '../file/service';
 import { Actor, Member } from '../member/entities/member';
 import { THUMBNAIL_FORMAT, THUMBNAIL_MIMETYPE, ThumbnailSizeFormat } from './constants';
 
+export const AVATAR_THUMBNAIL_PREFIX = 'avatars';
+export const ITEM_THUMBNAIL_PREFIX = 'thumbnails';
+
+@injectable()
 export class ThumbnailService {
-  fileService: FileService;
-  shouldRedirectOnDownload: boolean;
-  prefix: string;
+  private readonly fileService: FileService;
+  private prefix: string = ITEM_THUMBNAIL_PREFIX;
 
-  constructor(fileService: FileService, shouldRedirectOnDownload: boolean, prefix: string) {
-    this.shouldRedirectOnDownload = shouldRedirectOnDownload;
+  constructor(fileService: FileService) {
     this.fileService = fileService;
-    this.prefix = prefix ?? 'thumbnails';
   }
 
-  buildFilePath(itemId: string, name: string) {
+  public setPrefix(prefix: string) {
```

Comment:
It can be a setter function
```suggestion
  set prefix(prefix: string) {
```

Reply:
good catch ! 🙌